### PR TITLE
[vcpkg.cmake] fix X_VCPKG_APPLOCAL_DEPS_INSTALL

### PR DESCRIPTION
**Pull request by strega-nil**
This also does a script audit of this code

### CODE REVIEW

File: scripts/buildsystems/vcpkg.cmake [hunk 1]
```diff
@@ -602,29 +602,47 @@ endfunction()
 #
 # Note that this function requires CMake 3.14 for policy CMP0087
 function(x_vcpkg_install_local_dependencies)
-    if(Z_VCPKG_TARGET_TRIPLET_PLAT MATCHES "windows|uwp")
-        cmake_parse_arguments(PARSE_ARGV 0 __VCPKG_APPINSTALL "" "DESTINATION;COMPONENT" "TARGETS")
+    cmake_parse_arguments(PARSE_ARGV 0 arg
```
Comment (BillyONeal) [inline scripts/buildsystems/vcpkg.cmake:605]: I think somewhere else we said we want to work with cmake 3.1 but PARSE_ARGV was new in 3.7? Perhaps the version check should be moved to the front of the function.

**Review by BillyONeal — APPROVED** [review]

File: scripts/buildsystems/vcpkg.cmake (hunk 1 above)
Comment (dg0yt) [reply scripts/buildsystems/vcpkg.cmake:611]: In CMake 3.1, there was a module for that:
https://cmake.org/cmake/help/v3.1/module/CMakeParseArguments.html

**Pull request merged**